pkg/gui/files_panel.go

```diff
@@ -1,418 +1,910 @@
+	err = gui.refreshFiles()
+	if err != nil {
+		gui.Log.Errorf("Failed to refresh files at handleIgnoreFile: %s\n", err)
+		return err
+	}
+
+	return nil
 }
 
-func (gui *Gui) handleFileSelect(g *gocui.Gui, v *gocui.View) error {
-	file, err := gui.getSelectedFile(g)
+// handleFileSelect is called when a file is selected.
+// It checks if there are any changed files and if there is one
+// and it is selected, it gets rendered into the main view
+func (gui *Gui) handleFileSelect() error {
+
+	var content string
+
+	file, err := gui.getSelectedFile()
 	if err != nil {
+
 		if err != gui.Errors.ErrNoFiles {
+			gui.Log.Errorf("Failed to get selected file in handlefileselect: %s\n", err)
+			return err
+		}
+
+		err = gui.renderString(gui.g, "main", gui.Tr.SLocalize("NoChangedFiles"))
+		if err != nil {
+			gui.Log.Errorf("Failed to render string in handlefileselect: %s\n", err)
 			return err
 		}
-		gui.renderString(g, "main", gui.Tr.SLocalize("NoChangedFiles"))
-		return gui.renderfilesOptions(g, nil)
+
+		err = gui.renderGlobalOptions()
+		if err != nil {
+			gui.Log.Errorf("Failed to renderfilesoptions in handlefileselect: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
-	gui.renderfilesOptions(g, &file)
-	var content string
+
+	err = gui.renderGlobalOptions()
+	if err != nil {
+		gui.Log.Errorf("Failed to renderfilesoptions in handlefileselect: %s\n", err)
+		return err
+	}
+
 	if file.HasMergeConflicts {
-		return gui.refreshMergePanel(g)
+
+		err = gui.refreshMergePanel()
+		if err != nil {
+			gui.Log.Errorf("Failed to refreshmergepanel in handlefileselect: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
 
 	content = gui.GitCommand.Diff(file)
-	return gui.renderString(g, "main", content)
+
+	err = gui.renderString(gui.g, "main", content)
+	if err != nil {
+		gui.Log.Errorf("Failed to render string in handlefileselect: %s\n", err)
+		return err
+	}
+
+	return nil
 }
 
+// handleCommitPress is called when a user commits changes.
+// g and v are passed by the gocui library.
+// returns and error if something goes wrong.
 func (gui *Gui) handleCommitPress(g *gocui.Gui, filesView *gocui.View) error {
+
 	if len(gui.stagedFiles()) == 0 && !gui.State.HasMergeConflicts {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("NoStagedFilesToCommit"))
+
+		err := gui.createErrorPanel(gui.Tr.SLocalize("NoStagedFilesToCommit"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleCommitPress: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
-	commitMessageView := gui.getCommitMessageView(g)
-	g.Update(func(g *gocui.Gui) error {
-		g.SetViewOnTop("commitMessage")
-		gui.switchFocus(g, filesView, commitMessageView)
+
+	commitMessageView, err := gui.g.View("commitMessage")
+	if err != nil {
+		gui.Log.Errorf("Failed to get commitMessage view at handleCommitPress: %s\n", err)
+		return err
+	}
+
+	gui.g.Update(func(g *gocui.Gui) error {
+		_, err = g.SetViewOnTop("commitMessage")
+		if err != nil {
+
```

Comment:
I'm guessing that as this is still WIP these still need to be filled out :)

Reply:
haha i wish, totally forgot those, will add them asap